rust/agama-lib/src/manager/http_client.rs

```diff
@@ -0,0 +1,25 @@
+use crate::{base_http_client::BaseHTTPClient, error::ServiceError};
+
+pub struct ManagerHTTPClient {
+    client: BaseHTTPClient,
+}
+
+impl ManagerHTTPClient {
+    pub fn new() -> Result<Self, ServiceError> {
+        Ok(Self {
+            client: BaseHTTPClient::new()?,
+        })
+    }
+
+    pub fn new_with_base(base: BaseHTTPClient) -> Self {
+        Self { client: base }
+    }
+
+    pub async fn probe(&self) -> Result<(), ServiceError> {
+        // BaseHTTPClient did not anticipate POST without request body
+        let dummy_body: Vec<u8> = vec![];
+        self.client
+            .post_void("/manager/probe_sync", &dummy_body)
```

Comment:
Instead of a dummy `Vec<u8>`, you could use just the void type.

```suggestion
            .post_void("/manager/probe_sync", &())
```